trade-finance-manager-ui/component-tests/utilisation-reports/_macros/payment-details-table-row.component-test.ts

```diff
@@ -75,11 +80,71 @@ describe(component, () => {
       ...aPaymentDetailsTableRow(),
       [property]: 'Some custom property',
     };
-    const wrapper = getWrapper(paymentDetailsRow);
+    const wrapper = getWrapper({ paymentDetailsRow });
 
     wrapper.expectElement(`tr td[data-cy="${property}"]:contains("Some custom property")`).toExist();
   });
 
+  describe('when userCanEdit is set to true', () => {
+    const userCanEdit = true;
+
+    it.each([FEE_RECORD_STATUS.READY_TO_KEY, FEE_RECORD_STATUS.RECONCILED])(
+      'renders the payment amount as plain text when the fee record status is %s',
+      (status) => {
+        const paymentDetailsRow: PaymentDetailsTableRow = {
+          ...aPaymentDetailsTableRow(),
+          feeRecordPaymentGroupStatus: status,
+          payment: {
+            ...aPaymentDetailsPayment(),
+            amount: { formattedCurrencyAndAmount: 'GBP 123.45', dataSortValue: 0 },
+          },
+        };
+        const wrapper = getWrapper({ paymentDetailsRow, userCanEdit });
+
+        wrapper.expectElement('tr td:contains("GBP 123.45")').toExist();
+        wrapper.expectElement('tr a:contains("GBP 123.45")').notToExist();
+      },
+    );
+
+    it.each(difference(Object.values(FEE_RECORD_STATUS), [FEE_RECORD_STATUS.READY_TO_KEY, FEE_RECORD_STATUS.RECONCILED]))(
+      'renders the payment amount as a link to the edit payment page when the fee record status is %s',
+      (status) => {
+        const paymentDetailsRow: PaymentDetailsTableRow = {
+          ...aPaymentDetailsTableRow(),
+          feeRecordPaymentGroupStatus: status,
+          payment: {
+            ...aPaymentDetailsPayment(),
+            id: 24,
+            amount: { formattedCurrencyAndAmount: 'GBP 123.45', dataSortValue: 0 },
+          },
+        };
+        const wrapper = getWrapper({ reportId: 12, paymentDetailsRow, userCanEdit });
+
+        wrapper.expectElement('tr td:contains(a)').toExist();
+        wrapper.expectLink('td a').toLinkTo('/utilisation-reports/12/edit-payment/24', 'GBP 123.45');
+      },
+    );
+  });
+
+  describe('when userCanEdit is set to false', () => {
+    const userCanEdit = false;
+
```

Comment:
Should we add assertions for `it.each([FEE_RECORD_STATUS.READY_TO_KEY, FEE_RECORD_STATUS.RECONCILED])(...` when `userCanEdit` is false?

Reply:
We do - we have `it.each(Object.values(FEE_RECORD_STATUS))` which includes all the fee record statuses